File: Sep12/Lecture7Examples/Example27/MainActivity.java
package com.varunshankar.example27;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;
import android.os.Bundle;


public class MainActivity extends AppCompatActivity{
    private Fragment weatherFragment;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if(savedInstanceState!=null)
        {
            weatherFragment = getSupportFragmentManager().findFragmentByTag("weather_frag");
        }
        else {
            WeatherFragment weatherFragment = new WeatherFragment();
            FragmentTransaction fTrans = getSupportFragmentManager().beginTransaction();
            fTrans.replace(R.id.fl_frag_weather, weatherFragment, "weather_frag");
            fTrans.commit();
        }
    }
}

File: Sep12/Example27/app/src/main/java/com/varunshankar/example27/MainActivity.kt
package com.varunshankar.example27

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment

class MainActivity : AppCompatActivity() {
    private var weatherFragment: Fragment? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (savedInstanceState != null) {
            weatherFragment = supportFragmentManager.findFragmentByTag("weather_frag")
        } else {
            weatherFragment = WeatherFragment()
            val fTrans = supportFragmentManager.beginTransaction()
            fTrans.replace(R.id.fl_frag_weather, weatherFragment as WeatherFragment, "weather_frag")
            fTrans.commit()
        }
    }
}